JsonPreprocessor/CJsonPreprocessor.py

```diff
@@ -590,12 +590,14 @@ def __checkAndUpdateKeyValue(self, sInputStr: str) -> str:
 
       Nested param "${abc}['xyz']" -> "str(${abc}['xyz'])"
         '''
-        if re.search("\"\s*[0-9A-Za-z@#%&*+=\\:,;,?<>~().\-_\s]*\s*\${\s*[0-9A-Za-z\.\-_]+\s*}(\[+\s*.+\s*\]+)*\s*[0-9A-Za-z@#%&*+=\\:,;,?<>~().\-_\s]*\s*\"", sInputStr.lower()):
-            sInputStr = re.sub("(\${\s*[0-9A-Za-z\.\-_]+\s*}(\[+\s*.+\s*\]+)*)", "str(\\1)", sInputStr)
+        pattern1 = "\"\s*[0-9A-Za-z\u0080-\uFFFF@#%&*+=\\:,;,?<>~().\-_\s]*\s*\${\s*[0-9A-Za-z_]+[0-9A-Za-z\.\-_]*\s*}(\[+\s*.+\s*\]+)*\s*[0-9A-Za-z\u0080-\uFFFF@#%&*+=\\:,;,?<>~().\-_\s]*\s*\""
```

Comment:
Hi Son,
can you explain me why you don't do non greedy:
`"\"*.?\${\s*[0-9A-Za-z_]+[0-9A-Za-z\.\-_]*\s*}*.?\""`

Any reason not to do like this? This should find any $-parameter expression?

Thank you,
 Thomas

Reply:
Hello Thomas,

At the beginning, I came up with the idea that I want to avoid some special characters such as "$" or "{, }".
This morning I have just tested deeply and refactored this code block.

Thank you,
Son

Reply:
Hello Thomas,

I observe that multiple nested parameters in 1 string value currently doesn't work.
For example, "The value of variable 1 is ${var_1}, variable 2 is ${var_2}, and variable 3 is ${var_3} in config file", our jsonpreprocessor package could not replace ${var_1}, ${var_2}, and ${var_3} in the string.
I will implement it and push another commit to this pull-request.

Thank you,
Son

Reply:
Hi Son,
very good!
Either the regexp returns an Array of ${parameter} hits, or you resolve ${parameter} recursively till no hit is found in a string anymore.
@milanac030988 : Please support Son here with ideas how to solve this in an elegant way.
Thank you,
 Thomas

Reply:
Hello Thomas,

I have just refactored my code (quite a lot of changes) due to the new requirement.
I had carefully run my acceptance test (47 test cases of jsonpreprocessor) and I also executed acceptance test of Testsuitemanagement package. All test results are good now.

Thank you,
Son

Reply:
Hi Son,
very good! For me the regexps look better now. Shorter and less regexps reduce risks of bugs.
Nevertheless this features are not trivial to implement. I'm very aware about this.
Therefore I also wait for a feedback from @milanac030988 .
Thank you,
 Thomas

Reply:
Hi Son,
I make an immediate merge that Holger can proceed.
Based on Cuongs findings/proposals please still improve the codes.
Thank you,
 Thomas